libs/pubsub/pubsub.go

```diff
@@ -211,24 +209,20 @@ func (s *Server) Subscribe(ctx context.Context, clientID string, query Query, ou
 // returned to the caller if the context is canceled or if subscription does
 // not exist.
 func (s *Server) Unsubscribe(ctx context.Context, clientID string, query Query) error {
-	var origQuery Query
 	s.mtx.RLock()
 	clientSubscriptions, ok := s.subscriptions[clientID]
-	if ok {
-		origQuery, ok = clientSubscriptions[query.String()]
-	}
 	s.mtx.RUnlock()
 	if !ok {
 		return ErrSubscriptionNotFound
 	}
 
-	// original query is used here because we're using pointers as map keys
-	// ?
 	select {
-	case s.cmds <- cmd{op: unsub, clientID: clientID, query: origQuery}:
+	case s.cmds <- cmd{op: unsub, clientID: clientID, query: query}:
 		s.mtx.Lock()
-		// if its the only query left, should we also delete the client?
 		delete(clientSubscriptions, query.String())
+		if len(clientSubscriptions) == 0 {
+			delete(s.subscriptions, clientID)
+		}
```

Comment:
Do you think it is worth to add a test that covers this (new) behaviour? E.g. subscribe a query for client, call unsubscribe with that query&client, expect unsubscribeAll on that client to return an `ErrSubscriptionNotFound`?

Reply:
Good idea 👍